src/components/position.js

```diff
@@ -1,8 +1,8 @@
-var coordinatesMixin = require('../utils/coordinates').componentMixin;
 var registerComponent = require('../core/component').registerComponent;
-var utils = require('../utils/');
 
-module.exports.Component = registerComponent('position', utils.extend({
+module.exports.Component = registerComponent('position', {
+  type: 'coordinate',
```

Comment:
I've tried making coordinate values a single value rather than separate X, Y, Z values. It's a mess since you still want to be able to do `setAttribute('position', {x: 1, y: 2, z: 3})`. And that only addresses flat schemas. How would you specify it for the light component where it also has a `type`?

You could make a reserved `componentType` keyword or something. But that gets messy since every time you want to loop the schema for all the properties, you have to pop that keyword.

So the only solution is to nest the actual property definitions further which complicates things, and makes all the existing components not work.